src/operator.rs

```diff
@@ -850,6 +851,41 @@ impl UnaryPreAggFunction for ParseJson {
     }
 }
 
+#[derive(Clone)]
+pub struct ParseLogfmt {
+    input_column: Option<Expr>,
+}
+
+impl ParseLogfmt {
+    pub fn new(input_column: Option<String>) -> ParseLogfmt {
+        ParseLogfmt {
+            input_column: input_column.map(Expr::Column),
+        }
+    }
+}
+
+impl UnaryPreAggFunction for ParseLogfmt {
+    fn process(&self, rec: Record) -> Result<Option<Record>, EvalError> {
+        let pairs = {
+            let inp = get_input(&rec, &self.input_column)?;
+            // Record includes the trailing newline, while logfmt considers that part of the
+            // message if present. Trim any trailing whitespace.
+            logfmt::parse(&inp.trim_end())
+        };
+        let res = {
+            pairs.iter().fold(Some(rec), |record_opt, pair| {
+                record_opt.and_then(|record| match &pair.val {
+                    None => Some(record.put(&pair.key, data::Value::None)),
+                    Some(val) => {
+                        Some(record.put(&pair.key, data::Value::Str(val.to_string())))
```

Comment:
I'm not super format with logfmt semantics -- for angle-grinder, it'd probably be a bit more consistent with `parse` if you use `data::Value::from_str` which will attempt to upconvert numeric types.

Reply:
Everything is a string in logfmt. It's probably ok to try and upconvert.